runelite-client/src/main/java/net/runelite/client/plugins/barbarianassault/BarbarianAssaultPlugin.java

```diff
@@ -152,20 +154,32 @@ public void onGameTick(GameTick event)
 			return;
 		}
 
-		switch (client.getLocalPlayer().getPlayerComposition().getEquipmentId(KitType.CAPE))
+		for (Item item : client.getItemContainer(InventoryID.INVENTORY).getItems())
```

Comment:
This should probably be put into a new method `private Role getRole(ItemContainer itemContainer)`, and can definitely be removed from `onGameTick` to run instead on item container change.